Bio/SeqUtils/__init__.py

```diff
@@ -11,31 +11,136 @@
 
 
 import re
+import warnings
+from collections import Counter
 from math import pi, sin, cos
 
 from Bio.Seq import Seq, complement, complement_rna
 from Bio.Data import IUPACData
-
+from Bio import BiopythonDeprecationWarning
 
 ######################################
 # DNA
 ######################
 # {{{
 
+_ambiguous_gc_values = {
+    "A": 0,
+    "C": 1.0,
+    "T": 0,
+    "G": 1.0,
+    "S": 1.0,
+    "V": 2 / 3,
+    "B": 2 / 3,
+    "M": 0.5,
+    "R": 0.5,
+    "Y": 0.5,
+    "K": 0.5,
+    "X": 0.5,
+    "N": 0.5,
+    "H": 1 / 3,
+    "D": 1 / 3,
+    "W": 0,
+}
+
+
+def gc_content(seq: str, ambiguous: str = "ignore"):
+    """Calculate G+C percentage in seq (float between 0 and 1).
+
+    Copes with mixed case sequences. Ambiguous Nucleotides in this context are
+    those different from ATCGSW.
+
+    If ambiguous equals "ignore" (default) , it will treat only unambiguous
+    nucleotides (GCS) as counting towards the GC percentage.
+
+    If ambiguous equals "count", will use a "mean" value when counting the
+    ambiguous characters, for example, G and C will still be counted as 1, N
+    and X will be counted as 0.5, D will be counted as 0.33 etc. See
+    Bio.SeqUtils._ambiguous_gc_values for a full list.
+
+    If ambiguous equals "remove", will only count GCS towards the
+    percentage, but will also not count ambiguous characters towards the length
```

Comment:
Does this mean *other* ambiguous characters like R and N are not counted to the length (but S is)?

(I appreciate this text comes from the old function, that should get any clarifications made too)

Reply:
And W. It might be clearer to say it will only consider ACTGSW which are clearly A/T or G/C?

Reply:
Tried to make explicit all characters being considered as ambiguous in each context.